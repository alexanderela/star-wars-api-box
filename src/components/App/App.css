@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");

@font-face {
  font-family: "STARWARS";
  font-style:normal;
  font-weight:normal;
  src: 
      url("./swapi.ttf") format("truetype"),
      url("./swapi.svg#swapi") format("svg");
}

body {
  background: url('../../assets/SWAPI-background.jpg');
}

.App {
  display: flex;
  flex-direction: column;
  text-align: center;
}

.header {
  align-items: center;
  background: url('../../assets/SWAPI-background.jpg');
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  height: 20%;
  justify-content: center;
  position: relative;
  z-index: 1;
}

.app-title {
  color: #f9d500;
  font-family: "STARWARS";
  font-size: 10rem;
  margin: 3rem 0 0 0;
  -webkit-text-fill-color: transparent;
  -webkit-text-stroke-color: #f9d500;
  -webkit-text-stroke-width: 3px;
}

.title-i {
  -webkit-text-stroke-width: 7px;
}

.error-popup-placeholder {
  height: 2rem; 
  margin: 0 0 3rem; 
}

@media (max-width: 1060px) {
  .app-title {
    font-size: 10rem;
  }
}

@media (max-width: 940px) {
  .cat-button {
    font-size: 1.75rem;
    width: 9rem;
  }
}

@media (max-width: 690px) {
  .app-title {
    font-size: 8rem;
  }
  .title-i {
    -webkit-text-stroke-width: 3px;
  }
}

@media (max-width: 565px) {
  .cat-button {
    font-size: 1.75rem;
    width: 9rem;
  }
  .app-title {
    font-size: 7rem;
  }
}

@media (max-width: 530px) {
  .cat-button {
    font-size: 0.75rem;
    width: 5rem;
  }
}

@media (max-width: 480px) {
  .app-title {
    font-size: 6rem;
  }
}

@media (max-width: 430px) {
  .cat-button {
    width: 4rem;
  }
}

@media (max-width: 405px) {
  .app-title {
    font-size: 4.5rem;
  }
}

@media (max-width: 365px) {
  .cat-button {
    font-size: 0.5rem;
  }
}