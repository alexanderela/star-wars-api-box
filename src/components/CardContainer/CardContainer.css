.CardContainer {
  background: url('../../assets/SWAPI-background.jpg');
	border: 1px solid black;
  border-radius: 10px;
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat( auto-fit, minmax(17rem, 1fr) );
  margin: 0 5rem 3rem;
}

@media (max-width: 720px) {
  .CardContainer {
   margin: 0 auto 3rem
  }
}