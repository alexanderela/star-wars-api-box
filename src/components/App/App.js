import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import './App.css';
import Sidebar from '../Sidebar/Sidebar.js';
import CardContainer from '../CardContainer/CardContainer.js';
import DataCleaner from '../../DataCleaner.js';
import FavoriteButton from '../FavoriteButton/FavoriteButton.js';
import Nav from '../Nav/Nav.js';
import ErrorPopup from '../ErrorPopup/ErrorPopup.js';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dataCleaner: new DataCleaner(),
      films: {},
      people: [],
      vehicles: [],
      planets: [],
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false,
      favorites: [],
      scroll: true,
      favoritesSelected: false,
      showErrorPopup: false
    }
  }

  async componentDidMount() {
    this.getFilm()
    this.showPeople()
    this.showVehicles()
    this.showPlanets()
    this.getFavorites()
  }

  toggleFavorites = async (entry) => {
    let favorites;
    const { type, id } = entry
    const category = this.state[type].map(card => {
      if(card.id === id) {
        return {...card, isFavorite: !card.isFavorite}
      }
      return card
    })

    if (!this.isInFavorites(entry)) {
      favorites = [...this.state.favorites, entry]
    } else {
      favorites = this.state.favorites.filter(card => card.id !== id)      
    }

    await this.setState({ [type]: category, favorites: favorites })
    await this.setLocalStorage('favorites', favorites)
    await this.setLocalStorage( [type], category)
  }

  isInFavorites = (entry) => {
    return this.state.favorites.find((fav) => fav.id === entry.id)
  }

  toggleErrorPopup = () => {
    this.setState({
      showErrorPopup: !this.state.showErrorPopup
    })
  }

  showFavorites = () => {
    if(this.state.favorites.length) {
      this.setState({ 
        favoritesSelected: true,
        peopleSelected: false,
        planetsSelected: false,
        vehiclesSelected: false, 
      })
    } else {
        this.toggleErrorPopup()  
    }
  }

  getFilm = async () => {
    const films = await this.state.dataCleaner.getMovie()
    this.setState({ films })
  }

  getPeople = async () => {
    const people = await this.state.dataCleaner.getPerson()
    this.setState({ people })
    this.setLocalStorage('people', people)
  }

  getPlanets = async () => {
    const planets = await this.state.dataCleaner.getPlanet()
    this.setState({ planets })
    this.setLocalStorage('planets', planets)
  }

  getVehicles = async () => {
    const vehicles = await this.state.dataCleaner.getVehicle()
    this.setState({ vehicles })
    this.setLocalStorage('vehicles', vehicles)
  }

  getFavorites = async () => {
    const { favorites } = this.state
    if (!localStorage.favorites && favorites.length) {
      this.setLocalStorage('favorites', favorites)
    } else if (localStorage.favorites) {
      const newFavorites = await this.getLocalStorage('favorites')
      this.setState({favorites: newFavorites})
    }
  }

  setLocalStorage = (key, category) => {
    localStorage.setItem(key, JSON.stringify(category))
  }

  getLocalStorage = (categoryName) => {
    if(localStorage.length) {
      return JSON.parse(localStorage.getItem(categoryName))
    }
  }

  showPeople = async (e) => {  
    const { people, peopleSelected } = this.state
    if (!localStorage.people) {
      await this.getPeople()
    } else if (localStorage.people) {
      const retrievedPeople = this.getLocalStorage('people')
      this.setState({ 
        people: retrievedPeople,
        peopleSelected: true,
        planetsSelected: false,
        vehiclesSelected: false,
        favoritesSelected: false,
        scroll: false
      })
    }
  }
  
  showVehicles = async (e) => {
    const { vehicles, vehiclesSelected } = this.state
    if (!localStorage.vehicles) {
      await this.getVehicles()
    } else if (localStorage.vehicles) {
      const retrievedVehicles = this.getLocalStorage('vehicles')
      this.setState({
        vehicles: retrievedVehicles,
        vehiclesSelected: true,
        peopleSelected: false,
        planetsSelected: false,
        favoritesSelected: false,
        scroll: false
      })
    }
  }

  showPlanets = async (e) => {
    const { planets, planetsSelected } = this.state
    if (!localStorage.planets) {
      await this.getPlanets()
    } else if (localStorage.planets) {
      const retrievedPlanets = this.getLocalStorage('planets')
      this.setState({
        planets: retrievedPlanets,
        planetsSelected: true,
        peopleSelected: false,
        vehiclesSelected: false,
        favoritesSelected: false,
        scroll: false
      })
    }
  }

  render() {
    const { 
            films, 
            people, 
            vehicles, 
            planets, 
            peopleSelected, 
            planetsSelected, 
            vehiclesSelected, 
            favorites, 
            scroll,
            favoritesSelected,
            showErrorPopup
          } = this.state

    return (
      <div className="App">
        <header className="header">
          <h1 className="app-title">SWAP<span className="title-i">I</span> Box</h1>
          <Nav
            showPeople={this.showPeople}
            showPlanet={this.showPlanets}
            showVehicle={this.showVehicles}
            showFavorites={this.showFavorites}
            favorites={favorites}
            toggleErrorPopup={this.toggleErrorPopup}
          />
        </header>
        {showErrorPopup 
          ?
          <ErrorPopup
            text="close"
            closeError={this.toggleErrorPopup}
          />
          : <div className="error-popup-placeholder"></div>
        }
        <Route exact path="/" render={(props) => <Sidebar {...props} films={films} />}
        />
        <Route 
          exact path="/people" 
          render={(props) => <CardContainer {...props} 
            entries={people} 
            toggleFavorites={this.toggleFavorites} 
            favorites={favorites} 
          />}
        />
        <Route 
          exact path="/vehicles" 
          render={(props) => <CardContainer {...props} 
            entries={vehicles} 
            toggleFavorites={this.toggleFavorites} 
            favorites={favorites} 
          />}
        />
        <Route 
          exact path="/planets" 
          render={(props) => <CardContainer {...props} 
            entries={planets} 
            toggleFavorites={this.toggleFavorites} 
            favorites={favorites} 
          />}
        />
        <Route 
          exact path="/favorites" 
          render={(props) => <CardContainer {...props} 
            entries={favorites} 
            toggleFavorites={this.toggleFavorites} 
            favorites={favorites} 
          />}
        />
      </div>
    );
  }
}

export default App;

