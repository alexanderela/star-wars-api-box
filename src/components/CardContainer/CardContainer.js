import React from 'react';
import './CardContainer.css';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const CardContainer = ({ entries, toggleFavorites, favorites }) => {
  const cards = entries.map(entry => (
    <Card
      entry={entry}
      key={entry.name}
      toggleFavorites={toggleFavorites}
      favorites={favorites}
      isFavorite={entry.isFavorite}
    />
  ));

  return (
    <div className="CardContainer">
      { cards }
    </div>
  );
};

CardContainer.propTypes = {
  entries: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired,
};

export default CardContainer;
