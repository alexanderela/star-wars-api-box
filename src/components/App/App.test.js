/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import DataCleaner from '../../DataCleaner.js';
import CardContainer from '../CardContainer/CardContainer.js';
import Nav from '../Nav/Nav.js';
import films from '../../mockData/mockFilms.js';
import people from '../../mockData/mockPeople.js';
import vehicles from '../../mockData/mockVehicles.js';
import planets from '../../mockData/mockPlanets.js';
import resolvedPeople from '../../mockData/mockResolvedPeople.js';
import mockPerson from '../../mockData/mockPerson.js';
import mockSpecies from '../../mockData/mockSpecies.js';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <BrowserRouter>
  	<App />
  </BrowserRouter>, 
  div);
  ReactDOM.unmountComponentAtNode(div);
});

let wrapper;
let mockEvent;
let mockLuke;
let mockVehicle;
let mockPlanet;
let mockFilm;
let expected;
let mockEntry;
let	mockFavs = [{
          name: "Dina Caraballo",
          id: "Dina Caraballo",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        },
        {
          name: "Alex Ela",
          id: "Alex Ela",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        }, 
        {
          "name": "Luke Skywalker",
          "id": "Luke Skywalker",
          "isFavorite": true, 
          "type":"people",
          "properties": [
						{"header": 'Homeworld: ', "text": "Tatooine"},
						{"header": 'Species: ', "text": "human"},
						{"header": 'Population: ', "text": "200,000"},
						{"header": 'Language: ', "text": "Galactic Basic"}
						] 
	        }]

beforeEach(() => {
	wrapper = shallow(<App />);	
	mockEvent = { target: true }
})

afterEach(() => {
	localStorage.clear()
})

describe('App component', () => {
	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})

	it('has the correct default state for all categories', () => {
		expect(wrapper.state().films).not.toBe('{}')
		expect(wrapper.state().people).toEqual([])
		expect(wrapper.state().vehicles).toEqual([])
		expect(wrapper.state().planets).toEqual([])
		expect(wrapper.state().favorites).toEqual([])
	})

	it('passes the correct favoritesCount props to Nav', () => {
		wrapper.setState({ favorites: [{name: 'alvin', type: 'chipmunk'}] })
		const navComponent = wrapper.find(Nav)
		expect(navComponent.props().favorites.length).toEqual(1)
	})

	it('passes favoritesCount props to Nav if they exist in state', () => {
		const mockVehicles = [{"make": "mustang"}, {"make": "camaro"}]
		wrapper.setState({ favorites: mockVehicles })
		const navComponent = wrapper.find(Nav)
		expect(navComponent.props().favorites.length).toEqual(2)
	})

	describe('componentDidMount', () => {
		it('invokes corrects functions on componentDidMount', () => {
			wrapper.instance().getFilm = jest.fn()
			wrapper.instance().showPeople = jest.fn()
			wrapper.instance().showVehicles = jest.fn()
			wrapper.instance().showPlanets = jest.fn()
			wrapper.instance().getFavorites = jest.fn()
			wrapper.instance().componentDidMount()
			expect(wrapper.instance().getFilm).toHaveBeenCalled()
			expect(wrapper.instance().showPeople).toHaveBeenCalled()
			expect(wrapper.instance().showVehicles).toHaveBeenCalled()
			expect(wrapper.instance().showPlanets).toHaveBeenCalled()
			expect(wrapper.instance().getFavorites).toHaveBeenCalled()
		})
	})

	describe('toggleFavorites', () => {
		it('adds cards to favorites if not already contained in the favorites array', () => {
			mockEntry = {
		        "name": "Sand Cralwer",
		        "id": "Sand Cralwer",
		        "isFavorite": false, 
		        "type":"vehicles",
		        "properties": [
							{"header": 'Model: ', "text": "Digger Crawler"},
							{"header": 'Class: ', "text": "wheeled"},
							{"header": 'Passengers: ', "text": "1"}
						] 
		      }
			wrapper.setState({ favorites: []})
			wrapper.instance().toggleFavorites(mockEntry)
			expect(wrapper.state().favorites).toEqual([mockEntry])
		})

		it('removes cards from favorites if they are already in favorites array', async () => {
			expected = [{
          name: "Dina Caraballo",
          id: "Dina Caraballo",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        },
        {
          name: "Alex Ela",
          id: "Alex Ela",
          isFavorite: true, 
          type:"people",
          properties: [
						{header: 'Homeworld: ', text: "Earth"},
						{header: 'Species: ', text: "human"},
						{header: 'Population: ', text: "7,000,000,000"},
						{header: 'Language: ', text: "Galactic Basic"}
					] 
        }]
			mockEntry =	{
        "name": "Luke Skywalker",
        "id": "Luke Skywalker",
        "isFavorite": true, 
        "type":"people",
        "properties": [
					{"header": 'Homeworld: ', "text": "Tatooine"},
					{"header": 'Species: ', "text": "human"},
					{"header": 'Population: ', "text": "200,000"},
					{"header": 'Language: ', "text": "Galactic Basic"}
					] 
        }
			wrapper.setState({ favorites: mockFavs })
			wrapper.instance().toggleFavorites(mockEntry)
			expect(wrapper.state().favorites).toEqual(expected)
		})

	})

	describe('isInFavorites', () => {
		it('should find a match for the given entry', () => {
			mockEntry =	{
        "name": "Luke Skywalker",
        "id": "Luke Skywalker",
        "isFavorite": true, 
        "type":"people",
        "properties": [
					{"header": 'Homeworld: ', "text": "Tatooine"},
					{"header": 'Species: ', "text": "human"},
					{"header": 'Population: ', "text": "200,000"},
					{"header": 'Language: ', "text": "Galactic Basic"}
					] 
        }

      wrapper.setState({ favorites: mockFavs })
			expect(wrapper.instance().isInFavorites(mockEntry)).toEqual(mockEntry)
		})
	})

	describe('toggleErrorPopup', () => {
		it('should reverse the state of showError popup', () => {
			wrapper.instance().toggleErrorPopup()
			expect(wrapper.state().showErrorPopup).toBe(true)
		})
	})

	describe('showFavorites', () => {
		it('if there are favorites it should change favoritesSelected to true', () => {
			wrapper.setState({ favorites: mockFavs })
			wrapper.instance().showFavorites()
			expect(wrapper.state().favoritesSelected).toBe(true)
		})

		it('if there are no favorites it should call toggleErrorPopup', () => {
			const spy = spyOn(wrapper.instance(), 'toggleErrorPopup')
			wrapper.setState({ favorites: [] })
			wrapper.instance().showFavorites()
			expect(spy).toHaveBeenCalled()
		})
	})

	describe('getFilm', () => {
		beforeEach(() => {
			const mockDataCleaner = new DataCleaner
			wrapper.setState({ dataCleaner: mockDataCleaner})
		})

		it('should call getMovie', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getMovie')
			await wrapper.instance().getFilm()
			await expect(spy).toHaveBeenCalled()
		})

		it('should setState', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getMovie')
			expected = await wrapper.state().dataCleaner.getMovie()
			await wrapper.instance().getFilm()
			await expect(wrapper.state().films).toEqual(expected)
		})
	})

	describe('getPeople', () => {
		beforeEach(() => {
			const mockDataCleaner = new DataCleaner
			wrapper.setState({ dataCleaner: mockDataCleaner})
		})

		it('should call getPerson', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getPerson')
			await wrapper.instance().getPeople()
			await expect(spy).toHaveBeenCalled()
		})

		it('should setState', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getPerson')
			expected = await wrapper.state().dataCleaner.getPerson()
			await wrapper.instance().getPeople()
			await expect(wrapper.state().people).toEqual(expected)
		})
	})

	describe('getPlanets', () => {
		beforeEach(() => {
			const mockDataCleaner = new DataCleaner
			wrapper.setState({ dataCleaner: mockDataCleaner})
		})

		it('should call getPlanet', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getPlanet')
			await wrapper.instance().getPlanets()
			await expect(spy).toHaveBeenCalled()
		})

		it('should setState', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getPlanet')
			expected = await wrapper.state().dataCleaner.getPlanet()
			await wrapper.instance().getPlanets()
			await expect(wrapper.state().planets).toEqual(expected)
		})
	})

	describe('getVehicles', () => {
		beforeEach(() => {
			const mockDataCleaner = new DataCleaner
			wrapper.setState({ dataCleaner: mockDataCleaner})
		})

		it('should call getVehicle', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getVehicle')
			await wrapper.instance().getVehicles()
			await expect(spy).toHaveBeenCalled()
		})

		it('should setState', async () => {
			const spy = spyOn(wrapper.state().dataCleaner, 'getVehicle')
			expected = await wrapper.state().dataCleaner.getVehicle()
			await wrapper.instance().getVehicles()
			await expect(wrapper.state().vehicles).toEqual(expected)
		})
	})

	describe('setLocalStorage', () => {
		it('sets people data to local storage', () => {
			const mockData = [{ "class": "wheeled", "model": "Digger Crawler" }, 
			{"class": "tires", "model": "Mustang"}]
			wrapper.instance().setLocalStorage('mockData', mockData)
			expect(JSON.parse(localStorage.getItem('mockData'))).toEqual(mockData)
			localStorage.clear()
		})
	})

	describe('getLocalStorage', () => {
		it('gets and parses data from local storage', () => {
			localStorage.clear()
			const mockData = [{ "class": "wheeled", "model": "Digger Crawler" }, 
			{"class": "tires", "model": "Mustang"}]
			localStorage.setItem('mockData', JSON.stringify(mockData))
			const	getStorage = wrapper.instance().getLocalStorage('mockData')
			expect(getStorage).toEqual(mockData) 
			localStorage.clear()
		})
	})

	describe('showPeople', () => {
		it('sets people to state ', async () => {
			mockLuke = {
        "name": "Luke Skywalker",
        "id": "Luke Skywalker",
        "isFavorite": false, 
        "type":"people",
        "properties": [
					{"header": 'Homeworld: ', "text": "Tatooine"},
					{"header": 'Species: ', "text": "human"},
					{"header": 'Population: ', "text": "200,000"},
					{"header": 'Language: ', "text": "Galactic Basic"}
				] 
      }
			wrapper.state().dataCleaner.getPerson = jest.fn().mockImplementation(() => Promise.resolve(
		 		mockLuke))
			await wrapper.instance().showPeople()
		  await expect(wrapper.state().people).toEqual(mockLuke)
		})
		
		it('calls getVehicles if there are no vehicles in local storage', async () => {
			const spy = spyOn(wrapper.instance(), 'showVehicles')
			localStorage.clear()
			wrapper.instance().showVehicles()
			expect(spy).toHaveBeenCalled()
		})

		it('retrieves vehicles from local storage if there are vehicles in local storage', async () => {
			localStorage.setItem('vehicles', JSON.stringify(mockFavs))
			const spy = spyOn(wrapper.instance(), 'getLocalStorage')	
			wrapper.instance().showVehicles()	
			expect(spy).toHaveBeenCalled()				
		})
	})

	describe('showVehicles', () => {
		beforeEach(() => {
			mockVehicle = {
          "name": "Sand Cralwer",
          "id": "Sand Cralwer",
          "isFavorite": false, 
          "type":"vehicles",
          "properties": [
						{"header": 'Model: ', "text": "Digger Crawler"},
						{"header": 'Class: ', "text": "wheeled"},
						{"header": 'Passengers: ', "text": "1"}
					] 
        }
		})
		it('sets vehicles to state ', async () => {
			wrapper.state().dataCleaner.getVehicle = jest.fn().mockImplementation(() => Promise.resolve(
		 		mockVehicle))
			await wrapper.instance().showVehicles()
		  await expect(wrapper.state().vehicles).toEqual(mockVehicle)
		})

		it('calls getPeople if there are no people in local storage', async () => {
			const spy = spyOn(wrapper.instance(), 'showPeople')
			localStorage.clear()
			wrapper.instance().showPeople()
			expect(spy).toHaveBeenCalled()
		})

		it('retrieves people from local storage if there are people in local storage', async () => {
			localStorage.setItem('people', JSON.stringify(mockVehicle))
			const spy = spyOn(wrapper.instance(), 'getLocalStorage')	
			wrapper.instance().showPeople()	
			expect(spy).toHaveBeenCalled()				
		})
	})

	describe('showPlanets', () => {
		it('sets planets to state ', async () => {
			mockPlanet = {"name": "Alderaan", 
				"terrain": "grasslands, mountains", 
				"population": "2000000000", "climate": "temperate", 
			"residents": 
				["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"]}
			wrapper.state().dataCleaner.getPlanet = jest.fn().mockImplementation(() => Promise.resolve(
		 		mockPlanet))
			await wrapper.instance().showPlanets()
		  await expect(wrapper.state().planets).toEqual(mockPlanet)	
	})

		it('calls getPlanets if there are no planets in local storage', async () => {
			const spy = spyOn(wrapper.instance(), 'showPlanets')
			localStorage.clear()
			wrapper.instance().showPlanets()
			expect(spy).toHaveBeenCalled()
		})

		it('retrieves planets from local storage if there are planets in local storage', async () => {
			localStorage.setItem('planets', JSON.stringify(mockVehicle))
			const spy = spyOn(wrapper.instance(), 'getLocalStorage')	
			wrapper.instance().showPlanets()	
			expect(spy).toHaveBeenCalled()				
		})
	})
})