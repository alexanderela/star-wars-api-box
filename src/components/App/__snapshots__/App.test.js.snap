// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
        >
          <h1
            className="app-title"
          >
            SWAP
            <span
              className="title-i"
            >
              I
            </span>
             Box
          </h1>
          <Nav
            favorites={Array []}
            showFavorites={[Function]}
            showPeople={[Function]}
            showPlanet={[Function]}
            showVehicle={[Function]}
            toggleErrorPopup={[Function]}
          />
        </header>,
        <div
          className="error-popup-placeholder"
        />,
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/people"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/vehicles"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/planets"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/favorites"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="app-title"
            >
              SWAP
              <span
                className="title-i"
              >
                I
              </span>
               Box
            </h1>,
            <Nav
              favorites={Array []}
              showFavorites={[Function]}
              showPeople={[Function]}
              showPlanet={[Function]}
              showVehicle={[Function]}
              toggleErrorPopup={[Function]}
            />,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "SWAP",
                <span
                  className="title-i"
                >
                  I
                </span>,
                " Box",
              ],
              "className": "app-title",
            },
            "ref": null,
            "rendered": Array [
              "SWAP",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I",
                  "className": "title-i",
                },
                "ref": null,
                "rendered": "I",
                "type": "span",
              },
              " Box",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "favorites": Array [],
              "showFavorites": [Function],
              "showPeople": [Function],
              "showPlanet": [Function],
              "showVehicle": [Function],
              "toggleErrorPopup": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "error-popup-placeholder",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/people",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/vehicles",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/planets",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
          >
            <h1
              className="app-title"
            >
              SWAP
              <span
                className="title-i"
              >
                I
              </span>
               Box
            </h1>
            <Nav
              favorites={Array []}
              showFavorites={[Function]}
              showPeople={[Function]}
              showPlanet={[Function]}
              showVehicle={[Function]}
              toggleErrorPopup={[Function]}
            />
          </header>,
          <div
            className="error-popup-placeholder"
          />,
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/people"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/vehicles"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/planets"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/favorites"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="app-title"
              >
                SWAP
                <span
                  className="title-i"
                >
                  I
                </span>
                 Box
              </h1>,
              <Nav
                favorites={Array []}
                showFavorites={[Function]}
                showPeople={[Function]}
                showPlanet={[Function]}
                showVehicle={[Function]}
                toggleErrorPopup={[Function]}
              />,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "SWAP",
                  <span
                    className="title-i"
                  >
                    I
                  </span>,
                  " Box",
                ],
                "className": "app-title",
              },
              "ref": null,
              "rendered": Array [
                "SWAP",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I",
                    "className": "title-i",
                  },
                  "ref": null,
                  "rendered": "I",
                  "type": "span",
                },
                " Box",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "favorites": Array [],
                "showFavorites": [Function],
                "showPeople": [Function],
                "showPlanet": [Function],
                "showVehicle": [Function],
                "toggleErrorPopup": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "error-popup-placeholder",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/people",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/vehicles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/planets",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
